hri_safety_sense/src/JoystickHandler.cpp

```diff
@@ -15,93 +15,101 @@
 /**
  * ROS Includes
  */
-#include "ros/ros.h"
-#include "sensor_msgs/Joy.h"
+#include "rclcpp/rclcpp.hpp"
+#include "sensor_msgs/msg/joy.hpp"
 
 #include "hri_c_driver/VehicleMessages.h"
-#include "JoystickHandler.h"
+#include "JoystickHandler.hpp"
 
-using namespace hri_safety_sense;
-
-JoystickHandler::JoystickHandler(const std::string &frameId)
+hri_safety_sense::JoystickHandler::JoystickHandler(
+  rclcpp::node_interfaces::NodeTopicsInterface::SharedPtr nodeTopics,
+  rclcpp::node_interfaces::NodeLoggingInterface::SharedPtr nodeLogger,
+  rclcpp::node_interfaces::NodeClockInterface::SharedPtr nodeClock,
+  const std::string &frameId)
 {
-	// Joystick Pub
-	rawLeftPub = rosNode.advertise<sensor_msgs::Joy>("/joy", 10);
-	this->frameId = frameId;
+  // Joystick Pub
+  rawLeftPub = rclcpp::create_publisher<sensor_msgs::msg::Joy>(nodeTopics,
+    "/joy", 10);
+
+  this->nodeLogger = nodeLogger;
+  this->nodeClock = nodeClock;
+  this->frameId = frameId;
```

Comment:
Since this is the constructor, we can use [member initializer list](https://www.geeksforgeeks.org/when-do-we-use-initializer-list-in-c/) for these.

Reply:
d580097